kroxylicious-filters/kroxylicious-encryption/src/test/java/io/kroxylicious/filter/encryption/inband/InBandKeyManagerTest.java

```diff
@@ -210,14 +222,14 @@ void encryptionRetry() {
 
         var value = ByteBuffer.wrap(new byte[]{ 1, 2, 3 });
         var value2 = ByteBuffer.wrap(new byte[]{ 4, 5, 6 });
-        Record record = RecordTestUtils.record(value);
-        Record record2 = RecordTestUtils.record(value2);
+        Record record = RecordTestUtils.record(value, 0L);
```

Comment:
I found we need to start setting offsets to construct multi-record MemoryRecords from existing records, if they aren't monotonically increasing then it barfs when you try to build the MemoryRecords.

Reply:
I wonder... the RTU is specifically for tests.. perhaps there mileage in having factories which specifically return batches with non-contiguous offsets. Right now the defaults are kind-of dangerous.. they make is easy to write tests with built-in assumptions (e.g. that the offset is always zero, that offsets are contiguous, that batches are small, that the magic is always latest). What many of the tests actually need is coverage outside such defaults. So it feels like a higher-level builder would be useful.

Reply:
yeahp, the amount of combinations is already getting quite explosive in RTU, could imagine having a builder for Records or MemoryRecords would be a nice future PR.

Reply:
For consistency we should have the `offset` before the `value`.

Reply:
create #900 to capture RTU improvements